sktime/forecasting/trend/_spline_trend_forecaster.py

```diff
@@ -0,0 +1,223 @@
+
+from sktime.forecasting.trend import PolynomialTrendForecaster
+from sktime.forecasting.trend._util import _get_X_numpy_int_from_pandas
+
+
+class SplineTrendForecaster(PolynomialTrendForecaster):
+    r"""Forecast time series data with a spline trend.
+
+    Parameters
+    ----------
+    regressor : sklearn regressor estimator object, default=None
+        Define the regression model type. If not set, defaults to
+        sklearn.linear_model.LinearRegression.
+    n_knots : int, default=5
+        Number of knots of the splines if `knots` equals one of {'uniform', 'quantile'}.
+        Must be at least 2. Ignored if `knots` is array-like.
+    degree : int, default=1
+        Degree of the polynomial function.
+    knots : {'uniform', 'quantile'}or array-like of shape (n_knots, n_features),
+        default='uniform'
+        Determines knot positions such that first knot <= features <= last knot.
+        - 'uniform': `n_knots` are distributed uniformly between the
+        min and max values of the features.
+        - 'quantile': `n_knots` are distributed uniformly along the quantiles
+        of the features.
+        - array-like: Specifies sorted knot positions, including the boundary knots.
+        Internally, additional knots are added before the first knot and after
+        the last knot based on the spline degree.
+    extrapolation : {'error', 'constant', 'linear', 'continue', 'periodic'},
+        default='constant'
+        Determines how to handle values outside the min and max values of the
+        training features:
+        - 'error': Raises a ValueError.
+        - 'constant': Uses the spline value at the minimum or maximum feature as
+        constant extrapolation.
+        - 'linear': Applies linear extrapolation.
+        - 'continue': Extrapolates as is (equivalent to `extrapolate=True` in
+        `scipy.interpolate.BSpline`).
+        - 'periodic': Uses periodic splines with a periodicity equal to the distance
+        between the first and last knot, enforcing equal function values and
+        derivatives at these knots.
+    with_intercept : bool, default=True
+        If True, includes a feature in which all polynomial powers are
+        zero (i.e., a column of ones, acting as an intercept term in a linear
+        model).
+    prediction_intervals : bool, default=False
+        Whether to compute prediction intervals. If True, additional
+        calculations are done during fit to enable prediction intervals
+        to be calculated during predict. The prediction intervals are
+        based on an OLS regression model fitted to the data and calculated
+        according to Section 7.9 in [1]. Formulas are modified appropriately
+        if `with_intercept` is False.
+
+    References
+    ----------
+    .. [1] Hyndman, Rob J., and George Athanasopoulos. Forecasting: principles
+    and practice, 3rd edition. OTexts: Melbourne, Australia. OTexts.com/fpp3.
```

Comment:
Optional: While you should give original book reference, I think to help users you may give direct link to relevant section at https://otexts.com/fpp3/regression-matrices.html as well.